lib/flop/schema.ex

```diff
@@ -574,11 +575,12 @@ defimpl Flop.Schema, for: Any do
     join_fields = get_join_fields(opts)
     schema_fields = get_schema_fields(struct)
     alias_fields = Keyword.get(opts, :alias_fields, [])
+    custom_fields = get_custom_fields(opts)
 
-    all_fields = compound_fields ++ join_fields ++ schema_fields ++ alias_fields
+    all_fields = compound_fields ++ join_fields ++ schema_fields ++ alias_fields ++ custom_fields
 
     validate_no_duplicate_fields!(
-      compound_fields ++ join_fields ++ alias_fields
+      compound_fields ++ join_fields ++ alias_fields ++ custom_fields
```

Comment:
We'll need to add validation that custom fields are not set as sortable.

Reply:
I added the validation for now, but I think a later PR could possibly implement sorting for those custom fields as well, so renamed the `func` option to `filter`, to give some room for implementing this later if needed. 
Also I think it makes it more obvious that this MFA is only used for filtering.

Reply:
As I said, sorting won't be possible unless you override the complete sorting functionality of Flop.

Reply:
Couldn't we just make the custom fields behave like an alias field or let them sort on an existing schema field? 

In our case, with `inserted_at` for example sorting worked perfectly fine, since I just want to define custom filter logic...

Reply:
Alias fields can be referenced in an Ecto query, so Flop can sort by it. Custom fields cannot be referenced, so Flop cannot sort by it.

Reply:
I know that we can't do anything with our custom MFA for filtering when we are sorting, but in the case of typecasts this is not needed, since the db can sort without any typecast on the original field, right?

But if I name my custom_field like a field already present in the schema, or a field from a `selected_as` query, the field can be referenced by flop, no? 

```
sortable: [:inserted_at],
filterable: [:inserted_at],
custom_fields: [inserted_at: [filter: {CustomFiler, :date_filter, []}]
```
This could still work with the contents of `inserted_at`, since for sorting I don't need the typecasting. 
In fact this worked even before I added the validation, I think :).

And if we want to support aliasing as well we could later implement this:

Someone adds a `selected_as` to the query named `pet_count` and defines this:
```
custom_fields: [pet_count: [filter: {CustomFiler, :custom_pet_count, []}, sort: :alias]
```
the field could be sortable from the alias and filterable with the custom filter. 

Like this one could define a custom field, that is both sortable and filterable and completely customizable. The drawback is that then you have 2 ways to define alias fields, but I guess that is ok 🤷 

But it is all just ideas for a potential future implementation

Reply:
You can of course sort by the original field, or define a separate alias field and sort by that. I wouldn't go past that, as it would only make the API hard to comprehend.